library/Zend/Http/Client/Adapter/Curl.php

```diff
@@ -228,10 +228,12 @@ public function connect($host, $port = 80, $secure = false)
             curl_setopt($this->_curl, CURLOPT_CONNECTTIMEOUT, $this->_config['timeout']);
         }
 
-        if (defined('CURLOPT_TIMEOUT_MS')) {
-            curl_setopt($this->_curl, CURLOPT_TIMEOUT_MS, $this->_config['timeout'] * 1000);
-        } else {
-            curl_setopt($this->_curl, CURLOPT_TIMEOUT, $this->_config['timeout']);
+        if (array_key_exists('request_timeout', $this->_config)) {
+          if (defined('CURLOPT_TIMEOUT_MS')) {
+              curl_setopt($this->_curl, CURLOPT_TIMEOUT_MS, $this->_config['request_timeout'] * 1000);
+          } else {
+              curl_setopt($this->_curl, CURLOPT_TIMEOUT, $this->_config['request_timeout']);
+          }
```

Comment:
One last question: We have `CURLOPT_CONNECTTIMEOUT` listed in an instance variable entitled `$_invalidOverwriteableCurlOptions`. Do any of the following also need to be in that list?

- `CURLOPT_CONNECTTIMEOUT_MS`
- `CURLOPT_TIMEOUT`
- `CURLOPT_TIMEOUT_MS`

What's interesting is that `CURLOPT_CONNECTTIMEOUT` is in the list, but in this very method, we overwrite that value! As such, I'm a bit confused.